[CILogon] Don't make action a required field of CILogonOAuthenticator.allowed_idps follow-up

GeorgianaElena opened the pull request:
Follow-up to https://github.com/jupyterhub/oauthenticator/pull/516. Without this, it's still impossible to don't specify an `action`.

This also adds a test and documents this option.

Ref: https://github.com/jupyterhub/oauthenticator/issues/514

consideRatio reviewed (commented):
This will be 15.0.1 i figure, but right now the migration docs mentions only 15.0.0.

I suggest replacing 15.0.0 with 15.0.1 without further notes pretty much.

GeorgianaElena commented:
Thanks a lot @consideRatio! I believe I have addressed your comments.

After the merge of this PR, I believe we should go for a 15.0.1 release. I also opened https://github.com/jupyterhub/oauthenticator/pull/518, but maybe that needs more discussion and should be released later? What do you think?

consideRatio commented:
@GeorgianaElena yepp unless its another bugfix, let's not include it before 15.0.1 is released.

Want to go for a release? I merged a bigbucket bugfix pr just now as well btw.

GeorgianaElena commented:
> Want to go for a release? I merged a bigbucket bugfix pr just now as well btw.

Yep! Let's see how it goes. I'll open up a PR with the updated changelog shortly

consideRatio commented:
I agree!
